libsrc/leddevice/dev_net/LedDeviceYeelight.cpp

```diff
@@ -0,0 +1,1503 @@
+static const char CONFIG_TRANS_TIME [] = "transTime";
+static const char CONFIG_EXTRA_TIME_DARKNESS[] = "extraTimeDarkness";
+static const char CONFIG_DEBUGLEVEL [] = "debugLevel";
+
+static const char CONFIG_BRIGHTNESS_MIN[] = "brightnessMin";
+static const char CONFIG_BRIGHTNESS_SWITCHOFF[] = "brightnessSwitchOffOnMinimum";
+static const char CONFIG_BRIGHTNESS_MAX[] = "brightnessMax";
+static const char CONFIG_BRIGHTNESSFACTOR[] = "brightnessFactor";
+
+static const char CONFIG_RESTORE_STATE[] = "restoreOriginalState";
+
+static const char CONFIG_QUOTA_WAIT_TIME[] = "quotaWait";
+
+// Yeelights API
+static const int API_DEFAULT_PORT = 55443;
+static const quint16 API_DEFAULT_QUOTA_WAIT_TIME = 1000;
+
+// Yeelight API Command
+static const char API_COMMAND_ID[] = "id";
+static const char API_COMMAND_METHOD[] = "method";
+static const char API_COMMAND_PARAMS[] = "params";
+static const char API_COMMAND_PROPS[] = "props";
+
+static const char API_PARAM_CLASS_COLOR[] = "color";
+static const char API_PARAM_CLASS_HSV[] = "hsv";
+
+static const char API_PROP_NAME[] = "name";
+static const char API_PROP_MODEL[] = "model";
+static const char API_PROP_FWVER[] = "fw_ver";
+
+static const char API_PROP_POWER[] = "power";
+static const char API_PROP_MUSIC[] = "music_on";
+static const char API_PROP_RGB[] = "rgb";
+static const char API_PROP_CT[] = "ct";
+static const char API_PROP_COLORFLOW[] = "cf";
+static const char API_PROP_BRIGHT[] = "bright";
+
+// List of Result Information
+static const char API_RESULT_ID[] = "id";
+static const char API_RESULT[] = "result";
+//static const char API_RESULT_OK[] = "OK";
+
+// List of Error Information
+static const char API_ERROR[] = "error";
+static const char API_ERROR_CODE[] = "code";
+static const char API_ERROR_MESSAGE[] = "message";
+
+// Yeelight ssdp services
+static const char SSDP_ID[] = "wifi_bulb";
+static const char SSDP_FILTER[] = "yeelight(.*)";
+static const char SSDP_FILTER_HEADER[] = "Location";
+const quint16 SSDP_PORT = 1982;
+
+YeelightLight::YeelightLight( Logger *log, const QString &hostname, quint16 port = API_DEFAULT_PORT)
+	:_log(log)
+	  ,_debugLevel(0)
+	  ,_isInError(false)
+	  ,_host (hostname)
+	  ,_port(port)
+	  ,_tcpSocket(nullptr)
+	  ,_tcpStreamSocket(nullptr)
+	  ,_correlationID(0)
+	  ,_lastWriteTime(QDateTime::currentMSecsSinceEpoch())
+	  ,_lastColorRgbValue(0)
+	  ,_transitionEffect(YeelightLight::API_EFFECT_SMOOTH)
+	  ,_transitionDuration(API_PARAM_DURATION)
+	  ,_extraTimeDarkness(API_PARAM_EXTRA_TIME_DARKNESS)
+	  ,_brightnessMin(0)
+	  ,_isBrightnessSwitchOffMinimum(false)
+	  ,_brightnessMax(100)
+	  ,_brightnessFactor(1.0)
+	  ,_transitionEffectParam(API_PARAM_EFFECT_SMOOTH)
+	  ,_waitTimeQuota(API_DEFAULT_QUOTA_WAIT_TIME)
+	  ,_isOn(false)
+	  ,_isInMusicMode(false)
+{
+	_name = hostname;
+
+}
+
+YeelightLight::~YeelightLight()
+{
+	log (3,"~YeelightLight()","" );
+	if ( _tcpSocket != nullptr)
+	{
+		_tcpSocket->deleteLater();
```

Comment:
Is there a reason for not using ````delete```` directly ? Using ````deleteLater```` is something I need mostly when I need to delete the ````QObject::sender()```` object inside a slot. 

This will also do the job but there is a downside of using this method : It can make debugging harder in some specific cases as there will still be a code waiting to be executed even after the destructor finishes.

Reply:
See another possible pitfall : https://github.com/hyperion-project/hyperion.ng/pull/863/files#r449817735

Reply:
Removed deleteLater in various LedDevice files.